### DISCUSSION TITLE
Problem booting VMs with memory configs

### DISCUSSION THREAD

**Discussion by tk-nguyen** [cloud-hypervisor/cloud-hypervisor#2742] [General]
Hello, I'm running into a problem booting VM over the REST API socket. When my VM configuration includes the key `memory`, it doesn't boot, instead returning this error: `VmBoot(VmBoot(MemoryManager(SetUserMemoryRegion(SetUserMemory(Invalid argument (os error 22))))))`. Here's my VM configuration:
```json
{
        "memory": { "size": 1024 },
        "kernel": { "path": "/home/test/Downloads/hypervisor-fw" },
        "cmdline": { "args": "console=ttyS0 console=hvc0"},
        "disks": [{ "path": "/home/test/Downloads/focal-server-cloudimg-amd64.raw" }],
        "rng": { "src": "/dev/urandom" },
        "net": [{ "ip": "172.25.25.10", "mask": "255.255.255.0", "mac": "8F:98:8B:3E:57:1A" }]
}
```
When I remove the memory configs, it boots normally.

**Comment by sboeuf** [comment]
Hi @tk-nguyen , that's because your memory parameter is incorrect.
You need to provide `"memory": { "size": 1073741824 }` since I guess you're expecting a VM with 1G of RAM.

**Reply by tk-nguyen** [reply]
Ah yes, my bad. I've changed the memory value to that and it boots up as usual. I was reading the schema and expecting CH to accept the memory size in MB. Thank you!

**Reply by sboeuf** [reply]
Cool :)
I'm glad your issue is resolved!
